pkg/internal/codegen/parse/crd.go

```diff
@@ -84,10 +84,24 @@ func (b *APIs) parseCRDs() {
 					}
 
 					if HasStatusSubresource(resource.Type) {
-						subresources := &v1beta1.CustomResourceSubresources{
-							Status: &v1beta1.CustomResourceSubresourceStatus{},
+						if resource.CRD.Spec.Subresources == nil {
+							resource.CRD.Spec.Subresources = &v1beta1.CustomResourceSubresources{}
+						}
+						resource.CRD.Spec.Subresources.Status = &v1beta1.CustomResourceSubresourceStatus{}
+						resource.HasStatusSubresource = true
+					}
+
+					if HasScaleSubresource(resource.Type) {
+						if resource.CRD.Spec.Subresources == nil {
+							resource.CRD.Spec.Subresources = &v1beta1.CustomResourceSubresources{}
+						}
+						jsonPath := parseJSONPath(resource.Type)
+						labelSelctor := jsonPath["selectorpath"]
+						resource.CRD.Spec.Subresources.Scale = &v1beta1.CustomResourceSubresourceScale{
+							SpecReplicasPath:   jsonPath["specpath"],
+							StatusReplicasPath: jsonPath["statuspath"],
+							LabelSelectorPath:  &labelSelctor,
 						}
-						resource.CRD.Spec.Subresources = subresources
 						resource.HasStatusSubresource = true
```

Comment:
s/HasStatus/HasScale

Wondering if this is being used anywhere ?

Reply:
Good catch! This is heritage code from old kubebuilder version, will clean it.